airflow/www/static/js/dag/details/taskInstance/ExtraLinks.tsx

```diff
@@ -57,8 +57,27 @@ const ExtraLinks = ({
     if (!url) {
       return true;
     }
-    const urlRegex = /^(https?:)/i;
-    return urlRegex.test(url);
+    try {
+      const path = new URL(url);
+      // The URL creation will succeed for
+      // 1. Absolute URL
+      // 2. URL with javascript:()
+      // URL(url) will fail for relative paths.
+      // eslint-disable-next-line
+      if (path.protocol === "javascript:") {
+        return false; // javascript:() is not allowed
+      }
+      if (path.protocol === "http:" || path.protocol === "https:") {
+        return true; // Absolute URL are allowed
+      }
+    } catch (e: any) {
+      // Relative URL are handled here
+      const path = new URL(url, "http://localhost");
+      if (path.protocol === "http:") {
```

Comment:
Won't this always be true since we set localhost in the line above?

Reply:
Correct, updated the code, accordingly. PTAL.